vm/utils.c

```diff
@@ -0,0 +1,24 @@
+#include <stdio.h>
+#include <sys/stat.h>
+
+#include "tr.h"
+#include "internal.h"
+
+char *read_file_into_string(char *filename, int *str_len) {
```

Comment:
could we use size_t instead of int ?

Reply:
i think it'd be preferable to zero-initialize the str_len argument from inside this function, as opposed to the calling sites. do you agree ?

Reply:
we totally can use size_t, I'll do the change. We also can assign zero in the function, though we will need to declare str_len in the calling function and we will need to initialize it there to avoid compiler warning. So we will have duplicated initialization. Any concerns regarding this?